extensions/sensors/GetMovementSensors.cpp

```diff
@@ -0,0 +1,109 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <memory>
+#include <algorithm>
+#include <cctype>
+#include <cstdint>
+#include <cstring>
+#include <iostream>
+#include <iterator>
+#include <map>
+#include <set>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "utils/ByteArrayCallback.h"
+#include "core/FlowFile.h"
+#include "core/logging/Logger.h"
+#include "core/ProcessContext.h"
+#include "core/Relationship.h"
+#include "GetMovementSensors.h"
+#include "io/DataStream.h"
+#include "io/StreamFactory.h"
+#include "ResourceClaim.h"
+#include "utils/StringUtils.h"
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace processors {
+
+std::shared_ptr<utils::IdGenerator> GetMovementSensors::id_generator_ = utils::IdGenerator::getIdGenerator();
+
+const char *GetMovementSensors::ProcessorName = "EnvironmentalSensors";
+
+core::Relationship GetMovementSensors::Success("success", "All files are routed to success");
+
+void GetMovementSensors::initialize() {
+  logger_->log_trace("Initializing EnvironmentalSensors");
+  // Set the supported properties
+  std::set<core::Property> properties;
+
+  setSupportedProperties(properties);
+  // Set the supported relationships
+  std::set<core::Relationship> relationships;
+  relationships.insert(Success);
+  setSupportedRelationships(relationships);
+}
+
+void GetMovementSensors::onSchedule(const std::shared_ptr<core::ProcessContext> &context, const std::shared_ptr<core::ProcessSessionFactory> &sessionFactory) {
+  imu = RTIMU::createIMU(&settings);
```

Comment:
Does this createIMU alloc and give us owner? Didn't catch a delete in the destructor like the other one. If so, same comment applies w/r/t unique_ptr to give more of a RAII memory assurance given that nothing technically precludes a N call to onSchedule where N > 1.

Reply:
If memory serves me this was a singelton, but I updated the dependency version from a very old copy in the original PR, so I'll check and report back or fix. Thanks